### ISSUE TITLE
[ERROR] occlum-pal: Failed to create enclave with error code 0x2006: Invalid SGX device. Please make sure SGX module is enabled in the BIOS, and install SGX driver afterwards. (line 152, file src/pal_enclave.c)

### ISSUE THREAD

**Issue by niteeshkd** [confidential-containers/enclave-cc#297]
I was able to create enclave-cc in HW mode without attestation (i.e.  "security_validate": false for DECRYPT_CONFIG in enclave-cc/base/ccruntime-enclave-cc.yaml ) .  But, to make the enclave-cc works with attestation, i had to update the BIOS, enable MP registration, start MPA  registration. After doing these, when I retry to create the enclave-cc in HW mode without attestation, it complains with the following errors. 

```
$ kubectl describe pod enclave-cc-pod

Events:
  Type     Reason     Age                   From               Message
  ----     ------     ----                  ----               -------
  Normal   Scheduled  23m                   default-scheduler  Successfully assigned default/enclave-cc-pod to b77r44u11-node
  Normal   Pulled     22m                   kubelet            Successfully pulled image "docker.io/niteeshkd/occlum-hello-world:unenc" in 41.899s (41.899s including waiting)
  Normal   Created    22m                   kubelet            Created container hello-world
  Normal   Started    22m                   kubelet            Started container hello-world
  Warning  Failed     21m (x2 over 22m)     kubelet            Error: ImagePullBackOff
  Normal   Pulling    19m (x5 over 23m)     kubelet            Pulling image "docker.io/niteeshkd/occlum-hello-world:unenc"
  Warning  Failed     19m (x4 over 22m)     kubelet            Error: ErrImagePull
  Warning  BackOff    18m (x7 over 20m)     kubelet            Back-off restarting failed container hello-world in pod enclave-cc-pod_default(2aca3a17-bdcb-4820-9c35-6c484fac5990)
  Warning  Failed     8m14s (x7 over 22m)   kubelet            Failed to pull image "docker.io/niteeshkd/occlum-hello-world:unenc": rpc error: code = Internal desc = OCI config file already exists: "/run/enclave-cc/containers/occlum-hello-world_unenc/config.json"
  Normal   BackOff    3m18s (x59 over 22m)  kubelet            Back-off pulling image "docker.io/niteeshkd/occlum-hello-world:unenc"

$ kubectl logs enclave-cc-pod

[get_driver_type edmm_utility.cpp:116] Failed to open Intel SGX device. 
[open_se_device edmm_utility.cpp:178] Failed to open Intel SGX device. 
[get_driver_type /home/sgx/jenkins/ubuntuServer2004-release-build-trunk-222/build_target/PROD/label/Builder-UbuntuSrv20/label_exp/ubuntu64/linux-trunk-opensource/psw/urts/linux/edmm_utility.cpp:116] Failed to open Intel SGX device.
[ERROR] occlum-pal: Failed to create enclave with error code 0x2006: Invalid SGX device. Please make sure SGX module is enabled in the BIOS, and install SGX driver afterwards. (line 152, file src/pal_enclave.c)
```
I don't see any issue while testing the creation of SGX enclave using the sample code (e.g. github/linux-sgx/linux/installer/bin/sgxsdk/SampleCode/LocalAttestation/bin/app ) or a newly built SGX app in Occlum Dev Container.

**Comment by niteeshkd** [comment]
It seems the above issue is already [reported with occlum](https://github.com/occlum/occlum/issues/1318) .

**Comment by niteeshkd** [comment]
The above problem disappeared after reinstalling the SGX plugin in K8s cluster. We can close this issue.

**Comment by niteeshkd** [comment]
Closing as this is no  more issue for me.

**Comment by dcmiddle** [comment]
Thanks for researching and reporting.

**Comment by niteeshkd** [comment]
> any suggestions if we need to improve some docs on this? the enclave-cc quickstart briefly mentions the need for and "SGX enabled cluster" and the use of `sgx.intel.com/epc` resource request which implies the device plugin usage.

Yes, I would like to mention the following two things about the [SGX plugin REAME page](https://intel.github.io/intel-device-plugins-for-kubernetes/cmd/sgx_plugin/README.html).

1. There are three approaches (one with Operator and two with kubectl) to install SGX plugin. For me only the second approach (which deploys `NFD` and `NodeFaetureRules` before deploying `SGX plugin`) worked. The other two approaches did not report the value of  `sgx.intel.com/epc` . 
2. Regarding the above issue, after rebooting the machine, I noticed some error related to` SGX admission webhook` while installing `SGX plugin` using the second approach although I noticed the correct value of  sgx.intel.com/epc. Somehow, i managed to install `SGX plugin` after few trials of uninstalling/installing `SGX plugin` with `SGX admission webhook`.  Not sure whether `SGX admission webhook` should be installed before `NFD` / `NodeFaetureRules` or just before `SGX Plugin`. I would suggest one to make sure the SGX plugin is installed without any error apart from seeing the correct value of `sgx.intel.com/epc` . Also, it should be clearly mentioned that `SGX admission webhook` is required for both the approaches using kubectl.
